src/app/search/search-filters/sockets.tsx

```diff
@@ -245,6 +246,27 @@ const socketFilters: FilterDefinition[] = [
         );
     },
   },
+  {
+    keywords: 'enhancedperk',
+    description: tl('Filter.EnhancedPerk'),
+    format: 'range',
+    destinyVersion: 2,
+    filter: ({ filterValue }) => {
+      const compare = rangeStringToComparator(filterValue);
+      return (item: DimItem) => {
+        let enhancedPerkCount = 0;
+        if (item.sockets) {
+          // count enhanced perks
+          for (const socket of item.sockets.allSockets) {
+            if (socket.plugged) {
+              isEnhancedPerk(socket.plugged) && enhancedPerkCount++;
+            }
+          }
+        }
+        return compare(enhancedPerkCount ?? 0);
```

Comment:
`enhancedPerkCount` can't ever be `undefined` or `null`.

```suggestion
        return compare(enhancedPerkCount);
```